src/main/groovy/net/wasdev/wlp/gradle/plugins/LibertyMultiServerTasks.groovy

```diff
@@ -0,0 +1,350 @@
+            group 'Liberty'
+            dependsOn 'libertyCreate-' + server.name
+        })
+
+        addTaskRule('Pattern: libertyPackage-<Server Name>', 'libertyPackage', PackageTask, {
+            description 'Generates a WebSphere Liberty Profile server archive.'
+            logging.level = LogLevel.DEBUG
+            group 'Liberty'
+            dependsOn installDependsOn(server, 'libertyCreate')
+        })
+
+        addTaskRule('Pattern: libertyDump-<Server Name>', 'libertyDump', DumpTask, {
+            description 'Dumps diagnostic information from the Liberty Profile server into an archive.'
+            logging.level = LogLevel.INFO
+            group 'Liberty'
+        })
+
+        addTaskRule('Pattern: libertyJavaDump-<Server Name>', 'libertyJavaDump', JavaDumpTask, {
+            description 'Dumps diagnostic information from the Liberty Profile server JVM.'
+            logging.level = LogLevel.INFO
+            group 'Liberty'
+        })
+
+        addTaskRule('Pattern: deploy-<Server Name>', 'deploy', DeployTask, {
+            description 'Deploys a supported file to the WebSphere Liberty Profile server.'
+            logging.level = LogLevel.INFO
+            group 'Liberty'
+            dependsOn 'libertyStart-' + server.name
+        })
+
+        addTaskRule('Pattern: undeploy-<Server Name>', 'undeploy', UndeployTask, {
+            description 'Removes an application from the WebSphere Liberty Profile server.'
+            logging.level = LogLevel.INFO
+            group 'Liberty'
+            dependsOn 'libertyStart-' + server.name
+        })
+
+        addTaskRule('Pattern: cleanDirs-<Server Name>', 'cleanDirs', CleanTask, {
+            description 'Deletes files from some directories from the WebSphere Liberty Profile server'
+            logging.level = LogLevel.INFO
+            group 'Liberty'
+        })
+
+        addTaskRule('Pattern: configureArquillian-<Server Name>', 'configureArquillian', ConfigureArquillianTask, {
+            description "Automatically generates arquillian.xml for projects that use Arquillian Liberty Managed or Remote containers."
+            logging.level = LogLevel.INFO
+            group 'Liberty'
+            dependsOn 'libertyCreate', 'processTestResources'
+            skipIfArquillianXmlExists = project.configureArquillian.skipIfArquillianXmlExists
+            arquillianProperties = project.configureArquillian.arquillianProperties
+        })
+    }
+
+    void addTaskRule (String pattern, String name, Class taskType, Closure configureClosure) {
+        project.tasks.addRule(pattern) { String taskName ->
+            if (taskName.startsWith(name)) {
+                project.task(taskName, type: taskType, overwrite:true) {
+                    server = project.liberty.servers.getByName(taskName - "${name}-")
+                }.configure(configureClosure)
+            }
+        }
+    }
+
+    boolean tasksUpToDate (List<String> taskList) {
```

Comment:
what if we set the upper level's outputs to be the collection of servers' outputs